include/nccl_ofi_rdma.h

```diff
@@ -77,6 +84,16 @@ typedef struct nccl_net_ofi_rdma_mr_handle {
 /* Contents of ctrl message sent from receiver to sender to advertise
    destination buffer */
 typedef struct nccl_net_ofi_rdma_ctrl_msg {
+	/* Message type, must be NCCL_OFI_RDMA_MSG_CTRL */
+	nccl_ofi_rdma_msg_type_t type;
+
+	/* A comm identitifer that uniquely identifies the comm
+	 * on the receiver side */
+	uint64_t remote_comm_id;
+
+	/* Message sequence number */
+	uint16_t msg_seq_num;
```

Comment:
This is going to pack really poorly.  Enums are hard to properly size, but you have small type, unit64_t, uint16_t, uint64_t.  So you're going to have padding after type and after msg_seq_num.  I don't think we need more than 32 bits for comm_id, so you could have something like:
```
uint16_t type;
uint16_t msg_seq_num;
uint32_t remote_comm_id;
```
and have a reasonably good packing.

Reply:
makes sense. I will rearrange the header and make comm_id 32 bits. 
I can use `__attribute__((packed))` for the enum to make it a single byte, instead of casting it to `uint16_t `. Any objection on doing that?

Reply:
Add a static_assert() on size of struct for the ones that go over the wire so we can catch such padding issues. Nit: From a code-organization perspective, having structs that go over wire in a separate header is good to garner extra attention during code reviews and such.

Reply:
Moving the structs that go on the wire in a separate header turns out to be not-so-small change, since there are structs in those messages, which leads to circular includes. I would leave this to another PR.